examples/express-graphql-http-functions/context.ts

```diff
@@ -0,0 +1,47 @@
+import { v4 as uuidv4 } from 'uuid';
+
+export type UserData = { name: string };
+export type User = { id: string, } & UserData;
+
+/** This is an in-memory database, for illustrative purposes */
+const USERS_BY_ID: Record<string, { id: string, name: string }> = {};
+
+function userNotFoundError(id: string): Error {
+  return new Error(`User with ID "${id}" was not found`)
+}
+
+export class UserService {
+  async listUsers(): Promise<User[]> {
+    return Object.values(USERS_BY_ID)
+  }
+  
+  async getUser(id: string): Promise<User> {
+    const user = USERS_BY_ID[id];
+    if (!user) {
+      throw userNotFoundError(id)
+    }
+    return user;
+  }
+
+  async addUser(userData: UserData): Promise<User> {
+    const id = uuidv4();
+    const user = USERS_BY_ID[id] = { id, ...userData };
+    return user;
+  }
+
+  async removeUser(id: string): Promise<void> {
+    delete USERS_BY_ID[id];
+  }
+
+  async updateUser(id: string, userData: Partial<UserData>): Promise<User> {
+    const user = USERS_BY_ID[id];
+    if (!user) {
+      throw userNotFoundError(id)
+    }
+    return Object.assign(user, userData);
+  }
+}
+
+export type Context = {
+  userService: UserService;
```

Comment:
It would also be cool to have an example that demonstrates using [DataLoader](https://github.com/graphql/dataloader) pattern. We could do that in a separate example, or add that here.

Reply:
Makes sense. Let's put data loader in its own example. It's an important enough concept that having a focused example will be nice.